### PR TITLE
Refactor configuration into window

### LINKED ISSUE #382: Editor details menus opens themselves even if they've been hidden by the user
The editor details menu is shown every time you click an object which is quite annoying tbh. Why have a show/hide settings button then anyway?

### PR DESCRIPTION
**Pull request by HaSa1002**
I also made some adjustments to the buttons in the top left (from now on global menu). Most importantly making the jump station button a MenuButton and putting all of these buttons into a container.

The config window is instantiated every time and saves its values when pressing save. Cancel, close, or pressing escape won't save the values.
Speaking of escape... In order to fix escape really working the way we wanna have it working, I had to refactor the menu into its own script. So that's that. I took the chance and replaced `get_parent().get_parent().save_world()` with a signal, too.

Sorry for that huge boy of a commit. I am out of practice :sweat_smile:

Fixes #382

### CODE REVIEW

File: src/Editor/Scripts/pause_menu.gd [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+func _on_SaveAndExit_pressed():
+	emit_signal("save_requested")
+	get_tree().paused = false
+	LoadingScreen.load_main_menu()
+
+
+func _on_SaveWithoutExit_pressed():
+	get_tree().paused = false
+	LoadingScreen.load_main_menu()
```
Comment (DasCapschen) [inline src/Editor/Scripts/pause_menu.gd]: This isn't `SaveWithoutExit`, this is `ExitWithoutSave`

**Review by DasCapschen — CHANGES_REQUESTED** [review]
Haven't tried it, but the code mostly looks good. 
Thank you.

File: src/Editor/Scripts/pause_menu.gd (hunk 1 above)
Comment (HaSa1002) [reply src/Editor/Scripts/pause_menu.gd]: uh lol. I just copied the code and that is the name Jean gave it xD

Comment (HaSa1002) [reply src/Editor/Scripts/pause_menu.gd]: fixed

**Review by DasCapschen — APPROVED** [review]
lgtm

**Pull request merged**